### DISCUSSION TITLE
add_ldflags  -Wl,-Bsymbolic

### DISCUSSION THREAD

**Discussion by DanielKui** [xmake-io/xmake#4169] [Q&A]
我在我的target工程里
add_ldflags("-Wl,-Bsymbolic")

link阶段，并没有-Wl,-Bsymbolic 这个标志， 我的用法是不是不太对？

![image](https://github.com/xmake-io/xmake/assets/578118/01afd17a-5139-48f2-8069-a18b05395ea1)

link命令为：

/usr/bin/g++ -o HXApi/bin/libHXQuoteApi_x64.so build/.objs/HXQuoteApi/linux/x86_64/release/common/HXBaseComm.cpp.o build/.objs/HXQuoteApi/linux/x86_64/release/common/HXCommUtil.cpp.o build/.objs/HXQuoteApi/linux/x86_64/release/common/Compress.cpp.o build/.objs/HXQuoteApi/linux/x86_64/release/common/HXPtclPacket.cpp.o build/.objs/HXQuoteApi/linux/x86_64/release/common/HXUtil.cpp.o build/.objs/HXQuoteApi/linux/x86_64/release/common/HXProtobuf/HXQuote.pb.cc.o build/.objs/HXQuoteApi/linux/x86_64/release/common/HXProtobuf/HXConst.pb.cc.o build/.objs/HXQuoteApi/linux/x86_64/release/HXApi/HXQuoteApi/ChannelTcp.cpp.o build/.objs/HXQuoteApi/linux/x86_64/release/HXApi/HXQuoteApi/HXQuoteApi.cpp.o build/.objs/HXQuoteApi/linux/x86_64/release/HXApi/HXQuoteApi/HXQuoteApiDerived.cpp.o build/.objs/HXQuoteApi/linux/x86_64/release/common/des.c.o build/.objs/HXQuoteApi/linux/x86_64/release/common/minilzo.c.o -shared -m64 -fPIC -LThirdParty/lib64_2022 -L/home/kb/.xmake/packages/f/fmt/10.0.0/a66a7229101747f49a8bfbaef7469bb3/lib -L/usr/lib/x86_64-linux-gnu -L/home/kb/.xmake/packages/z/zeromq/4.3.4/4abccef0b5bc4ae98261d0607cddc0f0/lib -LTools/LogManager/bin -s -lprotobuf-lite -lACE -lquill -lfmt -llz4 -lzmq -lLogManager_x64 -lpthread

这里面没有-Wl,-Bsymbolic

**Comment by waruqi** [comment]
so 库用 add_shflags 。。另外 flags 有自动检测，如果检测不过会忽略，可以 {force = true} 强制设置
